### DISCUSSION TITLE
Deviation from autonomous flight

### DISCUSSION THREAD

**Discussion by Chilingyue** [bitcraze/discussions#903] [Q&A]
When controlling autonomous flight of unmanned aerial vehicles through Python scripts, the lateral deviation after takeoff is too large, resulting in loss of control

**Comment by krichardsson** [comment]
Could you tell us a bit more about your setup. What positioning system are you using?

**Comment by Chilingyue** [comment]
We use Loco positioning system with Tdoa3 mode

**Comment by krichardsson** [comment]
You must provide much more information, there is unfortunately no way we can help you based on what you have written so far. 
Please describe the setup in terms of anchor positions, the environment and so on. What is your script doing? What kind of mechanisms do you use for communication and control? What kind of errors do you observe?

The only advice I can give you at this point is to read the [documentation](https://www.bitcraze.io/documentation/repository/) and use the [tutorials](https://www.bitcraze.io/documentation/tutorials/).

**Reply by Chilingyue** [reply]
We set anchor points according to official documents and build a 2 * 2 * 2m space. After successfully locating the anchor point, the official test script (Python) was used for testing flight, but the flight results were unstable. We repeatedly ran the same script and got different results, even flying out of the experimental space, but the Crazyfly PC client showed no abnormalities in the anchor, although the client was closed while running the script.The following three images show the location details of anchor points and experimental scenarios. Video for flight testing
![QQ图片20230915205401](https://github.com/bitcraze/discussions/assets/130750514/45a56bb7-d8cc-45af-98b2-07cbab9b9fdb)
![QQ图片20230915205407](https://github.com/bitcraze/discussions/assets/130750514/5f25234b-49ac-4c6e-bb43-b4d418d8ea78)
![QQ图片20230915205417](https://github.com/bitcraze/discussions/assets/130750514/28daa072-373f-4350-970d-951ac4173289)

https://github.com/bitcraze/discussions/assets/130750514/3e59de5c-4a50-4bea-bdbc-9d0ef6571984

**Reply by Chilingyue** [reply]
#Here is the test script

```python
import logging
import time

import cflib.crtp
from cflib.crazyflie.syncCrazyflie import SyncCrazyflie
from cflib.positioning.motion_commander import MotionCommander

URI = 'radio://0/80/2M'

# Only output errors from the logging framework
logging.basicConfig(level=logging.ERROR)

if __name__ == '__main__':
    # Initialize the low-level drivers (don't list the debug drivers)
    cflib.crtp.init_drivers(enable_debug_driver=False)

    with SyncCrazyflie(URI) as scf:
        # We take off when the commander is created
        with MotionCommander(scf) as mc:
            print('fly!')
            time.sleep(1)

            # There is a set of functions that move a specific distance
            # We can move in all directions
            print('up0.2')
            mc.up(0.2)
            # Wait a bit
            time.sleep(1)
            
           
            exit(0)

            # We land when the MotionCommander goes out of scope
            print('Landing!')
```

**Comment by krichardsson** [comment]
Thanks for the info!

The first think to do, before attempting to fly, is to make sure the positioning is working. Open the Loco-tab in the client and move the Crazyflie around by hand and check that the position that is shown in the client seems to be right. If it is not, something is not set up properly.

Which mode are you using for the loco system, TWR, TDoA2 or TDoA3?
Are all anchors in the client indicated as green?

**Reply by Chilingyue** [reply]
Thanks for your reply. We move the drone and the coordinate information displayed in the client is correct. We followed the instructions in the help document and used TDoA3 mode, with all client positioning displayed in green. After flying again, the drone's flight situation remained the same as in the previous video, and it was still unable to follow the instructions.

**Reply by krichardsson** [reply]
Do you place the Crazyflie pointing along the X-axis of your coordinate system, [like in this picture](https://www.bitcraze.io/documentation/tutorials/getting-started-with-flying-using-lps/#restart-the-crazyflie)?

**Reply by amitx64** [reply]
@Chiliyu1 I am getting the same issue:

https://github.com/bitcraze/discussions/assets/79006087/c301f83b-463e-4a93-87e1-6aab1aff6b99

**Reply by StewenceNewton** [reply]
While using loco position system we have to measure the distance to all the anchors taking one point as origin. You can consider the centre as origin or one of the anchor corner as an origin then calculate  the distance as coordinate system(x,y,z) then write the distances to anchors using cfclient
